src/lsp/protocol.ml

```diff
@@ -796,20 +5380,216 @@ module References = struct
+         | [] ->
+             (match Ppx_yojson_conv_lib.(!) extra with
+              | _::_ ->
+                  Ppx_yojson_conv_lib.Yojson_conv_error.record_extra_fields
+                    _tp_loc (Ppx_yojson_conv_lib.(!) extra) yojson
+              | [] ->
+                  (match Ppx_yojson_conv_lib.(!) includeDeclaration_field
+                   with
+                   | Some includeDeclaration_value ->
+                       { includeDeclaration = includeDeclaration_value }
+                   | _ ->
+                       Ppx_yojson_conv_lib.Yojson_conv_error.record_undefined_elements
+                         _tp_loc yojson
+                         [((Ppx_yojson_conv_lib.poly_equal
+                              (Ppx_yojson_conv_lib.(!)
+                                 includeDeclaration_field) None),
+                            "includeDeclaration")]))))
+   | _ as yojson ->
+       Ppx_yojson_conv_lib.Yojson_conv_error.record_list_instead_atom _tp_loc
+         yojson : Ppx_yojson_conv_lib.Yojson.Safe.t -> referenceContext)
+and result_of_yojson =
+  (let _tp_loc = "src/lsp/protocol.ml.References.result" in
+   fun t -> list_of_yojson Location.t_of_yojson t : Ppx_yojson_conv_lib.Yojson.Safe.t
+                                                      -> result)
+let _ = params_of_yojson
+and _ = referenceContext_of_yojson
+and _ = result_of_yojson
+let rec yojson_of_params =
+  (function
+   | { textDocument = v_textDocument; position = v_position;
+       context = v_context } ->
+       let bnds : (string * Ppx_yojson_conv_lib.Yojson.Safe.t) list = [] in
+       let bnds =
+         let arg = yojson_of_referenceContext v_context in ("context", arg)
+           :: bnds in
+       let bnds =
+         let arg = yojson_of_position v_position in ("position", arg) :: bnds in
+       let bnds =
+         let arg = TextDocumentIdentifier.yojson_of_t v_textDocument in
+         ("textDocument", arg) :: bnds in
+       `Assoc bnds : params -> Ppx_yojson_conv_lib.Yojson.Safe.t)
+and yojson_of_referenceContext =
+  (function
+   | { includeDeclaration = v_includeDeclaration } ->
+       let bnds : (string * Ppx_yojson_conv_lib.Yojson.Safe.t) list = [] in
+       let bnds =
+         let arg = yojson_of_bool v_includeDeclaration in
+         ("includeDeclaration", arg) :: bnds in
+       `Assoc bnds : referenceContext -> Ppx_yojson_conv_lib.Yojson.Safe.t)
+and yojson_of_result =
+  (fun v -> yojson_of_list Location.yojson_of_t v : result ->
+                                                      Ppx_yojson_conv_lib.Yojson.Safe.t)
+let _ = yojson_of_params
+and _ = yojson_of_referenceContext
+and _ = yojson_of_result
+[@@@end]
 end
 
 (* DocumentHighlight request, method="textDocument/documentHighlight" *)
-module TextDocumentHighlight = struct
-  type params = TextDocumentPositionParams.t [@@deriving yojson { strict = false }]
+module TextDocumentHighlight = struct [@@@ocaml.warning "-39"]
```

Comment:
I'm surprised that this annotation is necessary only on very few modules that have deriving